cmd/logd/main.go

```diff
@@ -14,36 +14,39 @@ import (
 	"os/signal"
 	"path/filepath"
 
-	"github.com/deref/exo/cmdutil"
-	"github.com/deref/exo/logcol/api"
-	"github.com/deref/exo/logcol/server"
+	"github.com/deref/exo/logd/api"
+	"github.com/deref/exo/logd/server"
+	"github.com/deref/exo/util/cmdutil"
 )
 
 func main() {
+	ctx := context.Background()
+	ctx, done := signal.NotifyContext(ctx, os.Interrupt)
+	defer done()
+
 	paths := cmdutil.MustMakeDirectories()
 	cfg := &server.Config{
 		VarDir: paths.VarDir,
 		Debug:  true,
 	}
-	ctx := context.Background()
+
 	collector := server.NewLogCollector(ctx, cfg)
-	if err := collector.Start(ctx); err != nil {
-		cmdutil.Fatalf("starting collector: %w", err)
-	}
 
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, os.Interrupt)
-		<-c
-		collector.Stop(ctx)
-		os.Exit(0)
-	}()
+	{
```

Comment:
My first attempt at this was to do an error group or something like that to run the log collector and http server in parallel. In that world, they share a cancellation context, but that's not right! You actually want them nested, so that on interrupt, the http server shutsdown _first_, and then the collector shuts down. If you do it in parallel, you have a race condition where some http requests may attempt to access a collector that has already been shutdown.